src/include/OpenImageIO/detail/fmt.h

```diff
@@ -26,6 +26,7 @@
 
 // fmt 8.1 stopped automatically enabling formatting of anything that supports
 // ostream output. This breaks a lot! Re-enable this old behavior.
+// NOTE: fmt 10.0 removed this support entirely.
```

Comment:
How old of a version of fmt do we need to support?

Would it be possible to drop the "deprecated ostream" macro here now that you've implemented the missing items?

If I've understood correctly, the formatter functionality is not actually new, its just that we didn't implement the formatters before, right ?

Reply:
I don't know how far back we "need" to support. I think we can drop some old releases (in master / 2.5-to-be), but I'm not sure how far to push forward.

I'd need to do some archeology on fmt itself to see how far back it fully supports the format extensions (maybe all the way back, I'm just not sure), and also make a judgment call about whether dropping that macro might break anything people depend on -- for example, if people are using any of the utilities in strutil.h such as OIIO::print, format, or debug (which all wrap the fmt functionality), might they be subtly depending on this macro when dealing with types with ostream adapters but no formatter templates?

I want this PR to be fully backportable to solve the immediate problem of fmt 10.0 breaking our build. They we can separately tackle whether there is simplifications we can make in master but with some acceptable risk of breaking users of format/print if they haven't kept up with writing formatters.

Reply:
Makes sense. From my reading of the fmt release notes, the ostream_formatter which you are using here was only added in 9.0 (same time they deprecated the "automatic" `<<` handling). So I guess this is the oldest version we can support.